backend/src/services/encrypt/encrypt.service.ts

```diff
@@ -0,0 +1,20 @@
+import bcrypt from 'bcrypt';
+
+class Encrypt {
+  async comparePasswords(
+    passwordHash: string,
+    password: string,
+  ): Promise<boolean> {
+    let res = false;
+
+    bcrypt.compare(password, passwordHash, (_err, result) => {
+      if (result) {
+        res = true;
+      }
+    });
+
+    return res;
```

Comment:
`bcript.compare` returns a promise, so you can get rid of using callback and mutating some values inside passed callbacks
```suggestion
   return bcrypt.compare(password, passwordHash);
```